test/TestStableToken.sol

```diff
@@ -63,13 +70,15 @@ contract TestStableToken is
     }
 
     function mint(address to, uint256 amount) external onlyOwnerOrMinter {
-        if (totalSupply() + amount > maxSupply) revert ExceedsMaxSupply();
+        // pre-check so we use custom error
+        if (totalSupply() + amount > cap()) revert ExceedsCap();
+        // ERC20CappedUpgradeable::_mint will still enforce the cap as a safety
```

Comment:
Another way to do this would be to use `try/catch` and then check for the return error, in which case you'd emit a custom error instead.
Not sure if worth it though, as it would make things look more complex than they are..